test/test-api.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+// Testcase to produce NodeReport via API call
+if (process.argv[2] === 'child') {
+  const nodereport = require('../');
+  nodereport.triggerReport();
+} else {
+  const tap = require('tap');
+  const common = require('./common.js');
+  const fs = require('fs');
+  const spawn = require('child_process').spawn;
+
+  var child = spawn(process.execPath, [__filename, 'child']);
+  child.on('exit', function(code, signal) {
+    tap.plan(3);
+    tap.equal(code, 0, 'Process exited cleanly');
+    // Locate and validate the NodeReport
+    var report = common.locate(child.pid);
+    tap.ok(report, 'found report ' + report);
+    if (report) {
+      common.validate(tap, report);
+      // delete the report when done
```

Comment:
why? just makes debugging harder, its ok for tests to leave ephemra on disk, CI cleans it up afterwards